instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt

```diff
@@ -281,19 +273,195 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
+        val rerouteUpdate = mapboxNavigation.routesUpdates().first {
+            it.reason == RoutesExtra.ROUTES_UPDATE_REASON_REROUTE
+        }
         assertEquals(
-            listOf(
-                RerouteState.Idle,
-                RerouteState.FetchingRoute,
-                RerouteState.Interrupted,
-                RerouteState.Idle,
+            mockRouteReroute.routeWaypoints,
+            rerouteUpdate.navigationRoutes.first().waypoints!!.map { it.location() }
+        )
+        rerouteStateTransitionAssertion.assert()
+    }
+
+    @Test(timeout = 10_000)
+    fun reroute_is_not_cancelled_when_route_refreshed() = sdkTest {
+        val mapboxNavigation = createMapboxNavigation(3_000)
+        val mockRoute = RoutesProvider.dc_short_with_alternative(activity)
+        val mockRouteReroute = RoutesProvider.dc_short_with_alternative_reroute(activity)
+        val originLocation = mockRoute.routeWaypoints.first()
+        val originLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude()
+            longitude = originLocation.longitude()
+        }
+        mockLocationReplayerRule.loopUpdate(originLocationUpdate, times = 1)
+        val offRouteLocation = mockRouteReroute.routeWaypoints.first()
+        val offRouteLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = offRouteLocation.latitude()
+            longitude = offRouteLocation.longitude()
+        }
+
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        val requestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRoute.routeResponseJson,
+            expectedCoordinates = mockRoute.routeWaypoints,
+            relaxedExpectedCoordinates = false
+        )
+        mockWebServerRule.requestHandlers.add(requestHandler)
+        val rerouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRouteReroute.routeResponseJson,
+            expectedCoordinates = listOf(
+                offRouteLocation,
+                mockRoute.routeWaypoints.last()
             ),
-            rerouteStates
+            relaxedExpectedCoordinates = true
         )
+        val responseModifier = DelayedResponseModifier(4000) { it }
+        rerouteRequestHandler.jsonResponseModifier = responseModifier
+        mockWebServerRule.requestHandlers.add(rerouteRequestHandler)
+        val refreshHandler = MockDirectionsRefreshHandler(
+            testUuid = "jpCHHUC26qFOwISCNLjar2xmTfI6Dxd0qCHOoqwt_1VAlESNvsr7Zg==",
+            readRawFileText(
+                activity,
+                R.raw.route_response_dc_short_with_alternative_refresh_route_0
+            ),
+            acceptedGeometryIndex = 0
+        )
+        mockWebServerRule.requestHandlers.add(refreshHandler)
```

Comment:
There is a lot of duplicated code (in this test and the test above) and it doesn't really say anything about the test cases. I mean the setup of refresh handlers, etc. Can we extract it to a method so that it doesn't take up a lot of lines in the test itself?